modules/doi/includes/utilities.inc

```diff
@@ -6,15 +6,40 @@
  */
 
 /**
+ * Get the CrossRef for the given url.
+ *
+ * @param string $url
+ *   A DOI url, or a DOI number with or without the "doi:" prefix. If the $url
+ *   is a CrossRef ID passed in as doi:10.1016/j.tiv.2011.10.017 or
+ *   10.1111/eve.12339 it will not get changed by running it through parse_url.
+ *
+ * @return string
+ *   CrossRef ID string.
+ */
+function islandora_doi_get_crossref_id_from_url($url) {
+  // Allows for DOI to be entered in the following formats: 10.1111/eva.12339,
+  // http://dx.doi.org/10.1111/eva.12339, http://doi.org/10.1111/eva.12339,
+  // or doi:10.1111/eva.12339.
+  $doi_url = parse_url(trim($url));
+  // Remove "/" from the path if it's the first or last character in the path.
+  $doi_url['path'] = rtrim($doi_url['path'], '/');
+  $doi_url['path'] = ltrim($doi_url['path'], '/');
```

Comment:
A thought: could just use regular trim()? To ... haha ... *trim* down the number of lines here?
( •_•)
( •_•)>⌐■-■
(⌐■_■)

pinging @adam-vessey for pun approval